content/developers/package-guidelines.md

```diff
@@ -250,12 +250,13 @@ associated files.
         `GenomicRanges` package is listed in the Depends: field of
         `GenomicAlignments`.  It is unusual for more than three
         packages to be listed as 'Depends:'.
-      + **Suggests:** is for packages used in vignettes or examples,
-        or in conditional code. This includes examples that make use
-        of annotation and/or experiment packages (e.g., `TxDb*`).
-        In cases where an external one-off function is required,
-        the use of conditional code is best practice. External package
-        availability can be conditionally checked via `requireNamespace()`.
+      + **Suggests:** is for packages used in vignettes, examples,
+        and in conditional code. Commonly, annotation and experiment
+        packages (e.g., `TxDb*`) used in vignette and example code are
+        included in this field thus avoiding users a costly download.
+        In the case where an external one-off function is required for
+        package code, external package availability can be checked via
+        `if (!requireNamespace('extraPKG')) ... else stop(...)`.
```

Comment:
hmm, if you can't load the namespace then do something; if you can load the namespace stop? Maybe
```
nmspc = loadNamespace("extraPKG")
```
which fails with an error if the package is not available?

Reply:
I think the recommendation is something along the lines of:
```
if (!requireNamespace('extraPKG'))
    stop("Install 'extraPKG' to use this function")
```

Reply:
Just as a note, developers have more room to add their own error messages with the second option. It feels to me that calling `loadNamespace` does not do enough...
```r
nmspc <- loadNamespace("BlueJeans")
Error in loadNamespace("BlueJeans") : 
  there is no package called 'BlueJeans'
```